src/coreclr/jit/morph.cpp

```diff
@@ -1993,8 +2024,13 @@ void fgArgInfo::EvalArgsToTemps()
                     if (setupArg->OperIsCopyBlkOp())
                     {
                         setupArg = compiler->fgMorphCopyBlock(setupArg);
-#if defined(TARGET_ARMARCH) || defined(UNIX_AMD64_ABI)
+#if defined(TARGET_ARMARCH) || defined(UNIX_AMD64_ABI) || defined(TARGET_LOONGARCH64)
+#if defined(TARGET_LOONGARCH64)
+                        // For LoongArch64, the struct {float a; float b;} passed by float-registers.
+                        if ((lclVarType == TYP_STRUCT) && (curArgTabEntry->numRegs == 1))
+#else
                         if (lclVarType == TYP_STRUCT)
+#endif
```

Comment:
HFAs are also passed like that, so why does this code work as-is on ARM/64?

Reply:
the struct {int a; float b;} is also used a float register while the int member is passed integer register.

Reply:
the struct {int a; float b;} is also used a float register while the int member is passed integer register.